app/controllers/application_controller.rb

```diff
@@ -72,7 +72,7 @@ def flash_and_redirect_back_with_message(exception)
       end
     end
 
-    redirect_to :back
+    redirect_back(fallback_location: (request.referer || root_path))
```

Comment:
Hi @tarebyte, 

I found that `redirect_back` uses `request.referer` as the default redirect location. The argument `fallback_location` will only be used when the HTTP `Referer` is missing, which means that `request.referer` is `nil`. So I guess we can drop the `request.referer` here as it should always be `nil` when `fallback_location` is used. WDYT?

Please correct me if i'm wrong, thanks! 😄 

Reference:

- http://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back